app/src/main/java/com/example/javaopencommerce/catalog/ItemQueryFacade.java

```diff
@@ -73,19 +58,41 @@ private Pair<List<Long>, Integer> getFilteredResults(SearchRequest request) {
         .map(o -> parseLong(o.getString("_id")))
         .collect(toList());
 
-    return Pair.of(itemIds, totalElementsCount);
+    return Pair.of(productIds, totalElementsCount);
+  }
+
+  private List<ItemDto> sortItems(List<ItemDto> productDtos, SearchRequest request) {
+    String sortingType = request.getSortBy().toUpperCase() + "-" + request.getOrder().toUpperCase();
+    return productDtos.stream().sorted(pickItemComparator(sortingType)).toList();
+  }
+
+  private Comparator<ItemDto> pickItemComparator(String sortingType) {
+    switch (sortingType) {
+      case "VALUE-ASC" -> {
+        return Comparator.comparing(ItemDto::getValueGross);
+      }
+      case "VALUE-DESC" -> {
+        return Comparator.comparing(ItemDto::getValueGross).reversed();
+      }
+      case "NAME-DESC" -> {
+        return Comparator.comparing(ItemDto::getName).reversed();
+      }
+      default -> {
+        return Comparator.comparing(ItemDto::getName);
+      }
+    }
```

Comment:
And the extended switch, from Java 14 is so much more readable:
```suggestion
      return switch (sortingType) {
        case "VALUE-ASC" -> Comparator.comparing(ItemDto::getValueGross);
        case "VALUE-DESC" -> Comparator.comparing(ItemDto::getValueGross).reversed();
        case "NAME-DESC" -> Comparator.comparing(ItemDto::getName).reversed();
        default -> Comparator.comparing(ItemDto::getName);
      };
```